src/code/z_bgcheck.c

```diff
@@ -602,15 +616,15 @@ f32 BgCheck_RaycastFloorStatic(StaticLookup* lookup, CollisionContext* colCtx, u
 
 /**
  * Compute wall displacement on `posX` and `posZ`
- * sets `wallPolyPtr` to `poly` if `wallPolyPtr` is NULL or not a damage wall
+ * sets `wallPolyPtr` to `poly` if `wallPolyPtr` is NULL or doesn't have `surfaceData[0] & 0x08000000` set
```

Comment:
I'm pretty confident that this particular check, for whatever reason, is checking if the surface is a damage wall.

Reply:
this particular check yes, I think it prioritizes finding a damage wall so if Link is at a corner with a safe wall and a hurty wall, the hurty wall is picked and link gets damaged

But the flag itself is used for a bit more, including for stuff on floors